src/vmm/src/lib.rs

```diff
@@ -487,6 +488,22 @@ impl Vmm {
 
         Ok(())
     }
+    // Create and add a i8042 device to the VMM.
+    fn add_i8042_device(&mut self) -> Result<()> {
+        let reset_evt = EventFdTrigger::new(libc::EFD_NONBLOCK).map_err(Error::IO)?;
+        let i8042_device = Arc::new(Mutex::new(I8042Wrapper(I8042Device::new(
+            reset_evt.try_clone().map_err(Error::IO)?,
+        ))));
+        self.vm.register_irqfd(&reset_evt, 1)?;
+        let range = PioRange::new(PioAddress(0x064), 0x8).unwrap();
```

Comment:
The base of a device is where its range starts, i.e. 0x060, and its length should be 0x5 (0x60-0x64). [Here](https://wiki.osdev.org/%228042%22_PS/2_Controller) you can find the specification for i8042.